File: Java/src/BigO/SpaceComplexity.java
package BigO;

import java.util.Arrays;

public class SpaceComplexity {
    public static void main(String[] args) {

        int[] n = {1,2,3,4,5};
        boooo(n);

        String[] newArray = arrayOfHiNTimes(6);
        Arrays.stream(newArray).forEach(System.out::println);


    }
    //O(1)
    public static void boooo(int []n){
        for (int i = 0; i < n.length; i++) {
            System.out.println("booooo!");
        }
    }
    //O(n)
    public static String[] arrayOfHiNTimes(int n){
        String[] hiArray = new String[n];
        for (int i = 0; i < n; i++) {
            hiArray[i] = "hi";
        }
        return hiArray;
    }
}

File: JavaScript/BigO/spaceComplexity.js


function boooo (n ){
    for( let i=0; i< n.length; i++ ){
        console.log('Booooo')
    }

}


boooo ([1,2,3,4,5])
// O(1) space complexity


function arrayOfHiNTimes(n){

    let hiArray = [];
    for( let i=0; i< n; i++ ){
        hiArray[i] = "hi";
    }

    console.log(hiArray);
    return hiArray;

}

// O(n)
arrayOfHiNTimes(4);
